src/Symfony/Component/Uid/BinaryUtil.php

```diff
@@ -128,7 +128,7 @@ public static function hexToDateTime(string $time): \DateTimeImmutable
 
             if (self::TIME_OFFSET_BIN <= $time) {
                 $time = self::add($time, self::TIME_OFFSET_COM2);
-                $time[0] = $time[0] & "\x7F";
+                $time[0] &= "\x7F";
```

Comment:
AFAIK, this is not the same for string indexes. Would need double checking if this was fixed in recent versions of PHP.

Reply:
What is the string index here? (not the `[0]`, right?)

It is causing CI to fail, let's revert for now indeed

Reply:
yes it is, same for the other similar changes in the PR

Reply:
can you elaborate on that `string index` a bit? I'm not aware what is the underlying issue here.

Reply:
Sure: https://3v4l.org/GI1NW

Reply:
Rule marked as risky, at least: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7418
perhaps, with time we can have better detection if we touch index or not